packages/admin/cms-admin/src/dam/blocks/DamFileDownloadLinkBlock.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { DamFileDownloadLinkBlockData, DamFileDownloadLinkBlockInput } from "../../blocks.generated";
+import { CmsBlockContext } from "../../blocks/CmsBlockContextProvider";
+import { DamPathLazy } from "../../form/file/DamPathLazy";
+import { FileField } from "../../form/file/FileField";
+import { GQLDamFileDownloadLinkFileQuery, GQLDamFileDownloadLinkFileQueryVariables } from "./DamFileDownloadLinkBlock.generated";
+
+export const DamFileDownloadLinkBlock: BlockInterface<DamFileDownloadLinkBlockData, DamFileDownloadLinkBlockData, DamFileDownloadLinkBlockInput> = {
+    ...createBlockSkeleton(),
+
+    name: "DamFileDownloadLink",
+
+    displayName: <FormattedMessage id="blocks.damFileDownloadLink" defaultMessage="CMS Asset" />,
+
+    previewContent: (state) => (state.file ? [{ type: "text", content: state.file?.name }] : []),
+
+    category: BlockCategory.Media,
+
+    defaultValues: () => ({
+        openFileType: "DOWNLOAD",
+    }),
+
+    state2Output: (state) => ({
+        fileId: state.file?.id ?? undefined,
+        openFileType: state.openFileType,
+    }),
+
+    output2State: async (output, { apolloClient }: CmsBlockContext): Promise<DamFileDownloadLinkBlockData> => {
+        const ret: DamFileDownloadLinkBlockData = {
+            openFileType: output.openFileType,
+        };
+
+        if (output.fileId === undefined) {
+            return ret;
+        }
+
+        const { data } = await apolloClient.query<GQLDamFileDownloadLinkFileQuery, GQLDamFileDownloadLinkFileQueryVariables>({
+            query: gql`
+                query DamFileDownloadLinkFile($id: ID!) {
+                    damFile(id: $id) {
+                        id
+                        name
+                        fileUrl
+                    }
+                }
+            `,
+            variables: { id: output.fileId },
+        });
+
+        const { damFile } = data;
+
+        ret.file = {
+            id: damFile.id,
+            name: damFile.name,
+            fileUrl: damFile.fileUrl,
+        };
+
+        return ret;
+    },
+
+    definesOwnPadding: true,
+
+    AdminComponent: ({ state, updateState }) => {
+        return (
+            <>
+                <BlocksFinalForm<{
+                    openFileType: DamFileDownloadLinkBlockData["openFileType"];
+                    file?: DamFileDownloadLinkBlockData["file"];
+                }>
+                    onSubmit={(newValues) => {
+                        updateState({
+                            file: newValues.file ?? undefined,
+                            openFileType: newValues.openFileType,
+                        });
+                    }}
+                    initialValues={{
+                        file: state.file,
+                        openFileType: state.openFileType ?? "DOWNLOAD",
+                    }}
+                >
+                    {state.file !== undefined ? (
```

Comment:
nit: I'd invert this ternary operator. Usually having a negation (!) in the check makes it harder to understand